src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -287,27 +287,31 @@
    :value    comment})
 
 (defn- create-authority-key-identifier
-  [public-key issuer-dn serial critical]
+  [public-key issuer-dn serial cert critical]
   {:oid      authority-key-identifier-oid
    :critical (boolean critical)
    :value    {:public-key    public-key
               :serial-number (if (number? serial) (biginteger serial))
-              :issuer-dn     issuer-dn}})
+              :issuer-dn     issuer-dn
+              :cert          cert
+              :truncate-key false}})
 
 (schema/defn ^:always-validate authority-key-identifier :- SSLExtension
   "Create an `Authority Key Identifier` extension from a `PublicKey` object. The
   extension created by this function is intended to be passed into
   `sign-certificate` and `generate-certificate-request`, at which time the key's
   hash will be computed and stored in the resulting object."
+  ([cert]
+   (create-authority-key-identifier nil nil nil cert false))
```

Comment:
@justinstoller the [RFC](https://www.rfc-editor.org/rfc/rfc5280.html#section-4.2.1.1) has the following text that I think means this should be hardcoded:

>Conforming CAs MUST mark this extension as non-critical.